### DISCUSSION TITLE
Faker helpers

### DISCUSSION THREAD

**Discussion by jcbenck** [mockoon/mockoon#1217] [Help] [answered]
1. I'm attempting to use optional parameters with Faker helpers, specifically for the string.alpha method. According to the [documentation ](https://fakerjs.dev/api/string.html#alpha), I've crafted the following syntax:

`{{faker 'string.alpha' 'length.min=2' 'length.max=5' 'casing=lower'}}`

also tried something like: `{{faker 'string.alpha' min=2 max=5 casing=lower}}`

However, it doesn't seem to be working as expected. Could you help me identify any issues or suggest the correct syntax for utilizing optional parameters with the string.alpha Faker helper?

<hr>

2. I'm trying to use `{{faker 'date.anytime'}}` and according to the Faker doc I'm expecting something in this format:
`'2022-07-31T01:33:29.567Z'`

But what I got is `Fri Jul 26 2024 22:18:35 GMT-0300 (Horário Padrão de Brasília)` Could you help me identify any issues or suggest who can I get random date in the expected format?

**Comment by 255kb** [comment, marked as answer]
Hi, 

For no. 1, this kind of syntaxes `{{faker 'string.alpha' 'length.min=2' 'length.max=5' 'casing=lower'}}` will not work as you are passing strings to Handlebars. 
From what I see `{{faker 'string.alpha' 10}}` is working and generating strings like `HkeWErCOYF`.
But the `min` `max` properties should be inside a `length` object, which is not something that we anticipated (I'm not sure many of their helpers are working that way). We only support "first-level" named parameters as we use handlebars as the templating engine. Passing objects as a parameter is not possible without creating it with the `object` helper: `{{faker 'string.alpha' (object length=(object min=20 max=20))}}`. Which is less than ideal. 
This would work too: `{{faker 'string.alpha' (object length=6 casing='upper')}}`

It's a bit unfortunate that they are adding helpers with an object as the parameter. I will open an issue and see if we can improve the support in the future. 

For no. 2, Faker's helper is in fact returning a JS Date object. It's not formatted at all as they show in the documentation. So, using this helper in the body will convert it as a String which gives this format (default in JS). They don't provide any formatting option. Which means you would either have to use our [`dateFormat` helper](https://mockoon.com/docs/latest/templating/mockoon-helpers/#dateformat): `{{dateFormat (faker 'date.anytime') 'yyyy-MM-dd\'T\'HH:mm:ss.SSS\'Z\''}}`, or one of our helper like [`date`](https://mockoon.com/docs/latest/templating/mockoon-helpers/#date).

**Reply by jcbenck** [reply]
Thank you very much for the clarifications @255kb!
